File: abracadabra.py
import time
import sys

start = time.clock()

top_cap = int(sys.argv[1]) if len(sys.argv) > 1 else 100
print 'Running the searcher over %d iterations' % top_cap

for cycle in range(0, top_cap):
    count_found = total_lines = 0
    with open('input.txt', 'r') as fs:
        for line in fs:
            total_lines += 1
            if line.find('abracadabra') > -1:
                count_found += 1

plural = ''
if count_found == 0:
    verb = 'was no'
elif count_found == 1:
    verb = 'was a total of 1'
else:
    verb = 'were a total of %d' % count_found
    plural = 's'

print 'There %s abracadabra %s in the file' % (verb, plural)

print 'There were a total of %d lines to evaluate' % total_lines

end = time.clock()
diff = end - start

print 'The whole process took %f s' % diff

File: abracadabra.php
<?php
global $argv;
$start = microtime(true);

$top_cap = (isset($argv[1]) && !empty($argv[1])) ? $argv[1] : 100;
echo "Running the searcher over {$top_cap} iterations" . PHP_EOL;

foreach (range(0, $top_cap) as $cycle) {
    $contents = explode(PHP_EOL, file_get_contents('input.txt'));
    array_pop($contents); // remove the last (empty) element of the resulting array
    $count_found = $total_lines = 0;
    foreach ($contents as $line) {
        ++$total_lines;
        if (strpos($line, 'abracadabra') !== false) {
            ++$count_found;
        }
    }
}

switch ($count_found) {
    case 0:
        echo 'There was no abracadabra in the file' . PHP_EOL;
        break;

    default:
        switch ($count_found) {
            case 1:
                $verb = 'was';
                $quote = 'abracadabra';
                break;

            default:
                $verb = 'were';
                $quote = 'abracadabras';
        }

        echo "There {$verb} {$count_found} {$quote} in the file" . PHP_EOL;
}

echo "There were a total of {$total_lines} lines to evaluate" . PHP_EOL;

$end = microtime(true);
$diff = $end - $start;

echo "The whole process took {$diff} s" . PHP_EOL;
?>
